3949 - TablePaginated: Filter management + Admin users -> name - Export with Hidden filters

yaguzmang opened the pull request:
https://github.com/user-attachments/assets/d3b92c7a-ff13-46c9-b1b0-1b99fee4fa9a

yaguzmang committed:
3949 - Fix table body not showing when count is hidden

yaguzmang committed:
3949 - Fetch data with hidden filters

yaguzmang committed:
3949 - Export filtered data

yaguzmang committed:
3949 - Disable export if count is 0 (if active)

yaguzmang commented on src/client/components/TablePaginated/hooks/useFilterValues.tsx:
```diff
@@ -0,0 +1,35 @@
+  }, [activeFilterValues, hiddenFilterValues])
+
+  return allFilterValues
+}
```
The last thing missing would be to handle filters that are not hidden and have a default value (like having a prefilled name for the text input, for instance). However, that's not necessary for this issue.

minotogna replied:
Aren't all filters model stored in redux ? Making this hook redundant ?

yaguzmang replied:
The hidden filters are not stored in redux. My first approach to handle those was setting them in redux but it would cause a re fetch. Something like this:

https://github.com/user-attachments/assets/648cecb5-f23d-40f6-a65f-8ee2c59bbda8

So I decided to keep them out of the store. And also because those don't have inputs.

We can keep them out or we could add logic that prevents that refetch. Either works. What do you think?

minotogna replied:
I think we should have a consistent way of storing models. if filters are in redux, then all filter values must be in redux. one source of the data is important. otherwise it will be impossible to know where is the source of the filters.

minotogna commented on src/client/components/TablePaginated/TablePaginated.tsx:
```diff
@@ -61,7 +61,7 @@ const TablePaginated = <Datum extends object>(props: Props<Datum>) => {
         {(exportTable || withFilters) && (
           <div className="table-paginated-actions">
-            {exportTable && <ExportButton path={path} />}
+            {exportTable && <ExportButton counterActive={counter.show} filters={filters} path={path} />}
```
Why should counter.show matter here ? The export csv can be disabled if there's no data ?

yaguzmang replied:
I can do it by just looking if data is empty, but it will flicker a bit since we immediately reset the data before every fetch: 

https://github.com/user-attachments/assets/79deabd9-6897-40e0-bba5-30fa6dd9fe26

I can add the change if that's fine

yaguzmang replied:
And yes, the repository page is working. I also checked recent activity, history, and the other places TablePaginated is used. 
![image](https://github.com/user-attachments/assets/be04e6ca-5e6f-42ae-b09c-6840017a79dc)

minotogna replied:
> I can do it by just looking if data is empty, but it will flicker a bit since we immediately reset the data before every fetch:
> 
>  2024-10-01.08-18-58.mp4 
> 
> I can add the change if that's fine

yes ok, but what about data is empty and not fetching ?

yaguzmang replied:
Okay, we don't have the loading status on redux so I'll add it and use it here 👍

minotogna replied:
sure 👍

minotogna replied:
actually, let's just enable the csv button if count?.total > 0. let's see later other improvements . would that work now?

yaguzmang replied:
We don't fetch the count if `counter.show` is false, so the export would be disabled for tables that don't display the counter, as count would be undefined (that's why I passed the counterActive prop).

minotogna replied:
for now it's ok. let's see later if and when needed . what do you think ?

yaguzmang replied:
Sure 👍

minotogna reviewed (commented):
Is the links and repository page working ?

yaguzmang committed:
3949 - Set hidden filters in redux & use count to disable export

yaguzmang committed:
3949 - Remove unnecessary count check

The table will render the skeleton if the data is undefined,
and won't render anything if the data is an empty array, so
the count check in the parent is unnecessary.

minotogna commented on src/client/components/TablePaginated/hooks/useSetHiddenFilters.tsx:
```diff
@@ -0,0 +1,45 @@
+  path: string
+}
+
+export const useSetHiddenFilters = (props: Props): boolean => {
```
Can we have a TablePaginated.init method instead of this ? What do you think ?

yaguzmang replied:
Sure, sounds good. You mean something like this:
```
export const useInitTablePaginated = ({ filters, path }: Props): void => {
  const dispatch = useAppDispatch()
  const isInitialized = useIsTablePaginatedInitialized(path)

  useEffect(() => {
    if (!isInitialized) {
      dispatch(TablePaginatedActions.init({ filters, path }))
    }
  }, [dispatch, filters, isInitialized, path])
}
```
And handle the other logic in `TablePaginatedActions.init`

minotogna replied:
yes. but what do u mean by 'other logic' ?

yaguzmang replied:
Added, much cleaner now, thanks 👍 
By other logic I meant setting the defaultValues in state.

minotogna replied:
Nice 👍

yaguzmang committed:
3949 - Add TablePaginated init
